core/server/worker/src/main/java/alluxio/worker/block/ClientRWLock.java

```diff
@@ -33,8 +33,13 @@
   /** Total number of permits. This value decides the max number of concurrent readers. */
   private static final int MAX_AVAILABLE =
           Configuration.getInt(PropertyKey.WORKER_TIERED_STORE_BLOCK_LOCK_READERS);
-  /** Underlying Semaphore. */
-  private final Semaphore mAvailable = new Semaphore(MAX_AVAILABLE, true);
+  /**
+   * Underlying Semaphore.
+   * Convert to non-fair lock for the accidental hold read lock forever and write lock request
```

Comment:
```
Uses the unfair lock to prevent a read lock that fails to release from locking the block forever and thus blocking all the subsequent write access.
```